util/math_reference.h

```diff
@@ -15,22 +15,289 @@
+                                      cl::sycl::vec<float, N> b) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x, float y) { return isunordered(x, y); }, a, b);
+}
+template <int N>
+cl::sycl::vec<int64_t, N> isunordered(cl::sycl::vec<double, N> a,
+                                      cl::sycl::vec<double, N> b) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x, double y) { return isunordered(x, y); }, a, b);
+}
 
 /* one argument relational reference */
-int isfinite(float x);
-int isinf(float x);
-int isnan(float x);
-int isnormal(float x);
-int signbit(float x);
+int32_t isfinite(float x);
+int64_t isfinite(double x);
+template <int N> cl::sycl::vec<int32_t, N> isfinite(cl::sycl::vec<float, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x) { return isfinite(x); }, a);
+}
+template <int N>
+cl::sycl::vec<int64_t, N> isfinite(cl::sycl::vec<double, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x) { return isfinite(x); }, a);
+}
+int32_t isinf(float x);
+int64_t isinf(double x);
+template <int N> cl::sycl::vec<int32_t, N> isinf(cl::sycl::vec<float, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x) { return isinf(x); }, a);
+}
+template <int N> cl::sycl::vec<int64_t, N> isinf(cl::sycl::vec<double, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x) { return isinf(x); }, a);
+}
+int32_t isnan(float x);
+int64_t isnan(double x);
+template <int N> cl::sycl::vec<int32_t, N> isnan(cl::sycl::vec<float, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x) { return isnan(x); }, a);
+}
+template <int N> cl::sycl::vec<int64_t, N> isnan(cl::sycl::vec<double, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x) { return isnan(x); }, a);
+}
+int32_t isnormal(float x);
+int64_t isnormal(double x);
+template <int N> cl::sycl::vec<int32_t, N> isnormal(cl::sycl::vec<float, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x) { return isnormal(x); }, a);
+}
+template <int N>
+cl::sycl::vec<int64_t, N> isnormal(cl::sycl::vec<double, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x) { return isnormal(x); }, a);
+}
+int32_t signbit(float x);
+int64_t signbit(double x);
+template <int N> cl::sycl::vec<int32_t, N> signbit(cl::sycl::vec<float, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int32_t, float, N>(
+      [](float x) { return signbit(x); }, a);
+}
+template <int N> cl::sycl::vec<int64_t, N> signbit(cl::sycl::vec<double, N> a) {
+  return sycl_cts::math::run_rel_func_on_vector<int64_t, double, N>(
+      [](double x) { return signbit(x); }, a);
+}
+
+int any(signed char);
```

Comment:
It is not clear what are all these declarations.
If it is useful, explain in a comment.